src/main/java/com/metamx/emitter/service/ServiceEmitter.java

```diff
@@ -16,34 +16,45 @@
 
 package com.metamx.emitter.service;
 
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
 import com.metamx.common.lifecycle.LifecycleStart;
 import com.metamx.common.lifecycle.LifecycleStop;
 import com.metamx.emitter.core.Emitter;
 import com.metamx.emitter.core.Event;
 
 import java.io.IOException;
+import java.util.Map;
 
 public class ServiceEmitter implements Emitter
 {
-  private final String service;
-  private final String host;
+  private final ImmutableMap<String, String> serviceDimensions;
   private final Emitter emitter;
 
   public ServiceEmitter(String service, String host, Emitter emitter)
   {
-    this.service = service;
-    this.host = host;
+    this(service, host, emitter, ImmutableMap.<String, String>of());
+  }
+
+  public ServiceEmitter(String service, String host, Emitter emitter, Map<String, String> otherServiceDimensions)
+  {
+    this.serviceDimensions = ImmutableMap
+        .<String, String>builder()
+        .put("service", Preconditions.checkNotNull(service))
+        .put("host", Preconditions.checkNotNull(host))
+        .putAll(otherServiceDimensions)
```

Comment:
`Map<String, String>` can have `null` but if any key or value is `null` then `putAll` will fail. Suggest either accounting for that or putting a warning to this effect in the constructor javadoc